propagation/http_trace_context_propagator.go

```diff
@@ -0,0 +1,134 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package propagation
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"go.opentelemetry.io/api/trace"
+
+	"go.opentelemetry.io/api/core"
+	apipropagation "go.opentelemetry.io/api/propagation"
+)
+
+const (
+	supportedVersion  = 0
+	maxVersion        = 254
+	traceparentHeader = "traceparent"
+)
+
+type httpTraceContextPropagator struct{}
+
+var _ apipropagation.TextFormatPropagator = httpTraceContextPropagator{}
+
+func (hp httpTraceContextPropagator) Inject(ctx context.Context, supplier apipropagation.Supplier) {
+	sc := trace.CurrentSpan(ctx).SpanContext()
+	if sc.IsValid() {
+		h := fmt.Sprintf("%.2x-%.16x%.16x-%.16x-%.2x",
+			supportedVersion,
+			sc.TraceID.High,
+			sc.TraceID.Low,
+			sc.SpanID,
+			sc.TraceOptions)
+		supplier.Set(traceparentHeader, h)
+	}
+}
+
+func (hp httpTraceContextPropagator) Extract(ctx context.Context, supplier apipropagation.Supplier) core.SpanContext {
+	h := supplier.Get(traceparentHeader)
+	if h == "" {
+		return core.EmptySpanContext()
+	}
+
+	sections := strings.Split(h, "-")
+	if len(sections) < 4 {
+		return core.EmptySpanContext()
+	}
+
+	if len(sections[0]) != 2 {
+		return core.EmptySpanContext()
+	}
+	ver, err := hex.DecodeString(sections[0])
+	if err != nil {
+		return core.EmptySpanContext()
+	}
+	version := int(ver[0])
+	if version > maxVersion {
+		return core.EmptySpanContext()
+	}
+
+	if version == 0 && len(sections) != 4 {
+		return core.EmptySpanContext()
+	}
+
+	if len(sections[1]) != 32 {
+		return core.EmptySpanContext()
+	}
+
+	result, err := strconv.ParseUint(sections[1][0:16], 16, 64)
+	if err != nil {
+		return core.EmptySpanContext()
+	}
+	var sc core.SpanContext
+
+	sc.TraceID.High = result
+
+	result, err = strconv.ParseUint(sections[1][16:32], 16, 64)
+	if err != nil {
+		return core.EmptySpanContext()
+	}
+	sc.TraceID.Low = result
+
+	if len(sections[2]) != 16 {
+		return core.EmptySpanContext()
+	}
+	result, err = strconv.ParseUint(sections[2][0:], 16, 64)
+	if err != nil {
+		return core.EmptySpanContext()
+	}
+	sc.SpanID = result
+
+	opts, err := hex.DecodeString(sections[3])
+	if err != nil || len(opts) < 1 {
+		return core.EmptySpanContext()
+	}
+	sc.TraceOptions = opts[0]
+
+	if !sc.IsValid() {
+		return core.EmptySpanContext()
+	}
+
+	return sc
+}
+
+func (hp httpTraceContextPropagator) GetAllKeys() []string {
+	return nil
+}
+
+// HttpTraceContextPropagator creates a new text format propagator that propagates SpanContext
+// in W3C TraceContext format.
+//
+// The propagator is then used to create CarrierInjector and CarrierExtractor associated with a
```

Comment:
This paragraph sounds like a comment from the future. I see no notion of CarrierInjectors or CarrierExtractors in this PR.

Reply:
comment was from past ;-). Earlier I had CarrierInjectors and CarrierExtractors.
Removed it.